torch/fx/experimental/Partitioner.py

```diff
@@ -0,0 +1,145 @@
+        for node in self.nodes:
+            if node.op == 'placeholder':
+                input_nodes.append(node)
+        for partition in self.parents:
+            input_nodes += partition.get_output_nodes()
+        return input_nodes
+
+    def get_output_nodes(self) -> List[Node]:
+        """Output nodes are the nodes that without any user inside this partition."""
+        output_nodes: List[Node] = []
+        for node in self.nodes:
+            index = self.graph_module.graph.nodes.index(node)
+            user_indexes = GraphManipulation.get_all_users_of(self.graph_module, index)
+            user_nodes = set([self.graph_module.graph.nodes[i] for i in user_indexes])
+            #check if user nodes has an intersection with self.nodes
+            if not set(self.nodes).intersection(user_nodes):
+                output_nodes.append(node)
+        return output_nodes
+
+    def __str__(self) -> str:
+        if self.partition_id == 0:
+            return "root_partition"
+        line: str = 'partition id: ' + str(self.partition_id) + '\n'
+        line += 'all nodes in this partition: '
+        for node in self.nodes:
+            line += '(' + node.name + ':' + node.op + ') '
+        line += '\n'
+        line += 'parent partitions:'
+        for parent in self.parents:
+            if parent.partition_id == 0:
+                line += 'root_partition '
+            else:
+                line += 'partition_' + str(parent.partition_id) + ' '
+        line += '\n'
+        line += 'children partitions:'
+        for child in self.children:
+            line +=  str(child.partition_id) + ' '
+        line += '\n'
+        line += 'input nodes: '
+        input_nodes = self.get_input_nodes()
+        for node in input_nodes:
+            line += '(' + node.name + ':' + node.op + ') '
+        line += '\n'
+        line += 'output nodes: '
+        output_nodes = self.get_output_nodes()
+        for node in output_nodes:
+            line += '(' + node.name + ':' + node.op + ') '
+        return line
+
+class Partitioner:
+    def __init__(self) -> None:
+        """
+        After a partitioner is created,
+        it first check if multiple types of devices (backends) are involved.
+        So far, we assume that there is only one backend and one device with unlimited memory
+        """
+        self.partitions = []
+        self.devices = []
+        self.graph_module = None
+        self.node_to_partitions = {}
+
+    def partition_graph(self, fx_module: GraphModule, devices: List[str]) -> Dict[Partition, List[Partition]]:
+        """
+        Given the fx_moudle and devices, find the partitions, do the partitions,
+        and then return a dictionary representing the DAG of partitions
+        """
+        self.graph_module = fx_module
+        self.devices = devices
+        #Create a dummy partition as root.
+        self.root_partition = self.create_partition()
+        if len(self.devices) == 1:
```

Comment:
For now you don't need this check at all, you should still partition the graph if you are presented more than one device.